src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/conditions/ExclusionZonePdfExtract.kt

```diff
@@ -0,0 +1,103 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.conditions
+
+import org.apache.pdfbox.Loader
+import org.apache.pdfbox.io.RandomAccessReadBuffer
+import org.apache.pdfbox.pdmodel.PDDocument
+import org.apache.pdfbox.pdmodel.encryption.InvalidPasswordException
+import org.apache.pdfbox.rendering.PDFRenderer
+import org.apache.pdfbox.text.PDFTextStripper
+import org.slf4j.LoggerFactory
+import org.springframework.web.multipart.MultipartFile
+import java.awt.Image
+import java.awt.image.BufferedImage
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import javax.imageio.ImageIO
+
+data class ExclusionZonePdfExtract(
+  var thumbnailImage: ByteArray,
+  var fullSizeImage: ByteArray,
+  var description: String,
+) {
+  companion object {
+    fun fromMultipartFile(file: MultipartFile): ExclusionZonePdfExtract? = Loader.loadPDF(RandomAccessReadBuffer(file.inputStream)).use { pdfDoc ->
+      return ExclusionZonePdfExtractBuilder(pdfDoc).build()
+    }
+  }
+}
+
+class ExclusionZonePdfExtractBuilder(private var pdfDoc: PDDocument) {
+  fun build(): ExclusionZonePdfExtract? {
+    try {
+      val description = description()
+      val fullSizeImage = fullSizeImage()
+      val thumbnailImage = thumbnailImage(fullSizeImage)
+
+      if (!listOf(description, thumbnailImage, fullSizeImage).contains(null)) {
+        return ExclusionZonePdfExtract(
+          description = description!!,
+          fullSizeImage = fullSizeImage,
+          thumbnailImage = thumbnailImage,
+        )
+      }
+    } catch (io: IOException) {
+      log.error("IO error ${io.message}")
+    } catch (ipe: InvalidPasswordException) {
+      log.error("Invalid password ${ipe.message}")
+    } catch (iae: IllegalArgumentException) {
+      log.error("Illegal Argument ${iae.message}")
```

Comment:
I know the catch code was lift and shift but am a little concerned that it just swallows this info, 
I know you log it!

Just wonder if this should be here

```
   if (pdfExtract == null) {
      val reason = if (file.isEmpty) "Empty file" else "failed to extract the expected image map"
      throw ValidationException("Exclusion zone upload document - $reason")
    }
```

Reply:
This is replication of the existing code, however you are free to push up a counter PR with what you would do differently.